coconut/control/control.go

```diff
@@ -278,15 +279,30 @@ func CreateEnvironment(cxt context.Context, rpc *coconut.RpcClient, cmd *cobra.C
 	payloadData := new(ConfigurationPayload)
 	var wfPath string
 	if cmd.Flags().Changed("configuration") && len(configPayload) > 0 {
-		configPayloadDoc, err := os.ReadFile(configPayload)
-		if err != nil {
-			err = errors.New("cannot read configuration payload file")
-			return err
-		}
-		err = json.Unmarshal(configPayloadDoc, &payloadData)
-		if err != nil {
-			err = errors.New("cannot unmarshal configuration payload")
-			return err
+		if strings.HasSuffix(strings.ToLower(configPayload), ".json") {
+			configPayloadDoc, err := os.ReadFile(configPayload)
+			err = json.Unmarshal(configPayloadDoc, &payloadData)
+			if err != nil {
+				err = errors.New("cannot unmarshal local configuration payload")
+				return err
+			}
+		} else {
+			configPayloadSplit := strings.Split(configPayload, "/")
+			if len(configPayloadSplit) == 0 {
+				err = errors.New("configuration payload file is not valid")
+				return err
+			}
+			configPayloadUri := configPayloadSplit[len(configPayloadSplit)-1]
```

Comment:
This too might be problematic, because within their runtime subpath, components can do anything including create subtrees. In other words, both `foo` and `bar/quux` are valid inputs to `-c`, because both `o2/runtime/COG-v1/foo` and `o2/runtime/COG-v1/bar/quux` are valid paths (and should work assuming payloads exist at those keys). On the other hand, we a priori exclude Consul full paths (i.e. `consul://...`) because Apricot `GetRuntimeEntry` doesn't allow us to read from anywhere.